blog/fast-portfolio-sorts/index.qmd

```diff
@@ -0,0 +1,178 @@
+---
+title: "Fast Portfolio Sorts"
+author:
+  - name: Christoph Scheuch
+    url: https://www.tidy-intelligence.com/
+date: "2024-06-06"
+description: A benchmark of R approaches for efficient portfolio sorts
+categories: 
+  - R
+  - Portfolio Sorts
+  - data.table
+image: thumbnail.png
+image-alt: A winner's podium with three genderless robots celebrating, each holding a trophy. The robots have a sleek, modern design with neutral features. The podium is positioned in a modern office setting with bright colors, contemporary furniture, large windows letting in natural light, and vibrant decor. The first-place robot stands in the center, elevated above the second and third place robots on either side, all expressing joy. Created by DALL-E.
+---
+
+In the world of finance and investment management, working with large datasets containing portfolio holdings is a common task. One crucial operation that often needs to be performed on these datasets is portfolio sorts (e.g., see our chapter on [Univariate Portfolio Sorts](../../r/univariate-portfolio-sorts.qmd). While R offers a variety of approaches through its base functions and popular packages like `dplyr` and `data.table`, the performance of these methods can vary significantly, especially when dealing with large datasets. As a data analyst or portfolio manager, it's essential to understand the trade-offs between different sorting approaches in terms of execution time and memory usage. This blog post aims toshed light on this topic by benchmarking the performance of various sorting methods in R, specifically for the use case of sorting portfolio holdings.
```

Comment:
"to shed light"

Reply:
I see where you are coming from, but my gut feeling is that no portfolio manager relies on fast computation of portfolio sorts for any financial application (I mean, this is not HF stuff).
I'd frame the paragraph as there are many ways of implementing standard tasks in R, for example, portfolio sorts using `base,` `dplyr,` or `dtplyr.` For researchers and data analysts, two points are crucial: The implementation should be correct and fast. The latter point is relevant even if portfolio sorts are a simple task. However, in light of the many proposed sorting variables (and due to methodological uncertainty), many applications are required to perform many different ways to sort portfolios. For that reason, computational efficiency can become critical.